src/main/webapp/script.js

```diff
@@ -21,12 +21,13 @@ function loadIndex() {
   // Determine whether all the questions should be fetched or just the ones that match the search.
   const fullTextSearch = (new URL(document.location)).searchParams.get("search");
   if (fullTextSearch === "1") {
-    // Fetch just the questions related to the string input in the search bar.
-    const stringSearchInput = (new URL(document.location)).searchParams.get("stringSearchInput");
-    searchQuestion(stringSearchInput);
+    // Fetch first page of the questions related to the string input in the search bar.
+    const stringSearchInput = 
+        (new URL(document.location)).searchParams.get("stringSearchInput");
+    searchQuestion(stringSearchInput, 1);
   } else {
-    // Fetch the whole forum.
-    fetchQuestions('forum');
+    // Fetch the whole forum on the first page.
+    fetchForum(1);
```

Comment:
It's worth adding a comment like 'fetchForm(/*pageNumber=*/1)' so it's more clear what the integer being passed in represents! Same elsewhere.

Reply:
done